reference/reference.go

```diff
@@ -195,40 +204,139 @@ func ParseNamed(s string) (Named, error) {
 	return named, nil
 }
 
-// WithName returns a named object representing the given string. If the input
-// is invalid ErrReferenceInvalidFormat will be returned.
-func WithName(name string) (Named, error) {
```

Comment:
It's not identical to `ParseNamed` because this function won't accept tags or digests. There are some cases where an item being parsed is strictly a name only, and tags or digests shouldn't be accepted. It would be useful to keep this convenience function to support this case. It's fine with me if it's implemented as `ParseNamed + NamedOnly`.